src/libsyntax/parse/parser.rs

```diff
@@ -4391,6 +4391,10 @@ impl<'a> Parser<'a> {
         self.token.is_keyword(keywords::Extern) && self.look_ahead(1, |t| t != &token::ModSep)
     }
 
+    fn is_existential_type_decl(&self) -> bool {
+        self.token.is_keyword(keywords::Existential) && self.look_ahead(1, |t| t.is_ident())
```

Comment:
Why check if the next token is an ident rather than checking if it is the keyword `type`?

Reply:
because I got confused and panicked by beating the problem with a shovel?

For real: I didn't find a `token::Type` and decided "oh checking for the concrete identifier is too hard"... all while i did that before the `&&`...

I'll fix it asap